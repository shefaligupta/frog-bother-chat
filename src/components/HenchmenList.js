import React from 'react';
import Button from '@material-ui/core/Button';

/**
 * HenchmenList class.
 * @param {Function} setHenchman
 * @param {string} man
 * @returns {*}
 * @constructor
 */
const HenchmenList = ({setHenchman, man}) => {
  const allHenchmen = ['Sonny', 'Fredo', 'Michael'];
  return (
      <div className="henchman">
        {
          allHenchmen.map((henchman) => (
              <div key={`Henchman-${henchman}`} className="button-henchmen">
                <Button
                    variant="contained"
                    className={man === henchman ? 'active-button' : 'select-henchman'}
                    disableElevation
                    onClick={() => setHenchman(henchman)}
                >
                  {henchman}
                </Button>
              </div>
          ))
        }
      </div>
  );
};

export default HenchmenList;
