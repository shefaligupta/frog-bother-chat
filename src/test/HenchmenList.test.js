import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import HenchmenList from '../components/HenchmenList';

it('should snapshot henchmen list', () => {
  const tree = renderer.create(<HenchmenList/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render Henchmen names button', () => {
  const {container} = render(<HenchmenList/>);
  const henchmanList = container.querySelector('[class="henchman"]');
  expect(henchmanList.children).toHaveLength(3);
});
