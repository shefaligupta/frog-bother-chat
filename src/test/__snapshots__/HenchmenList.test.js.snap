// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot henchmen list 1`] = `
<div
  className="henchman"
>
  <div
    className="button-henchmen"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained select-henchman MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sonny
      </span>
    </button>
  </div>
  <div
    className="button-henchmen"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained select-henchman MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Fredo
      </span>
    </button>
  </div>
  <div
    className="button-henchmen"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained select-henchman MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Michael
      </span>
    </button>
  </div>
</div>
`;
